src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -447,40 +449,41 @@ static Config::ModelType getModelType(const std::shared_ptr<const Model>& model)
            Config::ModelType::CNN : Config::ModelType::Unknown;
 }
 
-static Config::SnippetsMode getSnippetsMode(const std::map<std::string, std::string>& modelConfig, const Config& engineConfig) {
+static Config::SnippetsMode getSnippetsMode(const ov::AnyMap& modelConfig, const Config& engineConfig) {
     const auto& snippetsMode = modelConfig.find(InferenceEngine::PluginConfigInternalParams::KEY_SNIPPETS_MODE);
-    if (snippetsMode == modelConfig.end()) // not set explicitly
-        return Config::SnippetsMode::Enable; // enable by default
+    if (snippetsMode == modelConfig.end())    // not set explicitly
+        return Config::SnippetsMode::Enable;  // enable by default
 
-    const auto& val = snippetsMode->second;
-    if (val == PluginConfigInternalParams::IGNORE_CALLBACK)
+    const auto& val = snippetsMode->second.as<std::string>();
+    if (val == InferenceEngine::PluginConfigInternalParams::IGNORE_CALLBACK)
         return Config::SnippetsMode::IgnoreCallback;
-    else if (val == PluginConfigInternalParams::DISABLE)
+    else if (val == InferenceEngine::PluginConfigInternalParams::DISABLE)
         return Config::SnippetsMode::Disable;
     else
-        IE_THROW() << "Wrong value for property key SNIPPETS_MODE. Expected values: ENABLE/DISABLE/IGNORE_CALLBACK";
+        OPENVINO_THROW("Wrong value for property key SNIPPETS_MODE. Expected values: ENABLE/DISABLE/IGNORE_CALLBACK");
 }
 
-InferenceEngine::IExecutableNetworkInternal::Ptr
-Engine::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork &network, const std::map<std::string, std::string> &orig_config) {
-    OV_ITT_SCOPED_TASK(itt::domains::intel_cpu, "Engine::LoadExeNetworkImpl");
+std::shared_ptr<ov::ICompiledModel>
+Engine::compile_model(const std::shared_ptr<const ov::Model>& model, const ov::AnyMap& orig_config) const{
+    OV_ITT_SCOPED_TASK(itt::domains::intel_cpu, "Engine::compile_model");
     CREATE_DEBUG_TIMER(debugLoadTimer);
 
     // verification of supported input
-    for (const auto &ii : network.getInputsInfo()) {
-        auto input_precision = ii.second->getPrecision();
-
-        using hash_t = std::hash<typename std::underlying_type<Precision::ePrecision>::type>;
-
-        static const std::unordered_set<Precision::ePrecision, hash_t> supported_precisions = {
-            Precision::U8,   Precision::I8,
-            Precision::U16,  Precision::I16,
-            Precision::U32,  Precision::I32,
-            Precision::U64,  Precision::I64,
-            Precision::BF16, Precision::FP16,
-            Precision::FP32, Precision::FP64,
-            Precision::BOOL
-        };
+    for (const auto &ii : model->inputs()) {
+        auto input_precision = ii.get_element_type();
+        static const std::set<ov::element::Type_t> supported_precisions = {ov::element::Type_t::u8,
```

Comment:
Why was a binary tree chosen instead of a hash table?

Reply:
A error occurred when "ov::element::Type_t" was used for hash table, it seems hard to fix in a easy way, maybe we can find a better way to avoid this error.